go-controller/pkg/cni/helper_linux.go

```diff
@@ -413,7 +408,15 @@ func (pr *PodRequest) deletePodConntrack() {
 
 // PlatformSpecificCleanup deletes the OVS port
 func (pr *PodRequest) PlatformSpecificCleanup() error {
-	ifaceName := pr.SandboxID[:15]
+	var ifaceName string
+	if pr.CNIConf.DeviceID != "" {
+		// SR-IOV case
+		ifaceNames, _ := ovsFind("Interface", "name", "external-ids:sandbox="+pr.SandboxID)
```

Comment:
there is a slight chance that somehow the VF representor will be removed/renamed, in that case ovsFind will fail to find the Interface. 
in this case, i think you should catch the error and log it.

In addition, if we do hit this case we will have an OVS port without an interface right ? are you relying on the `checkForStaleOVSInterfaces()` to remove the port from the bridge ?

Reply:
@adrianchris unfortunately we can't depend on checkForStaleOVSInterfaces() since that funciton only checks for `ofport` that is set to `-1`. 

If cmdDel() fails to remove the port from `br-int`, then the `ofport` will not be `-1`. It will be a non-negative number and will not be cleared by checkForStaleOVSInterfaces()

Reply:
I think you refer to a different (perfectly valid) case where the interface exists in the interface table but the delete port command below fails.

I was referring to a scenario where the ovsFind fails to find the interface, meaning the ofport is set to -1.

1. connect vf rep to bridge
```
# ovs-vsctl add-port br-int pf1vf5
# ovs-vsctl -- --columns=name,ofport list Interface 
....
....                                                                                                                                                                                                                                                                                                                                                          
name                : pf1vf5                                                                       ​
ofport              : 7 
```

2. rename vf rep
```
# ip link set pf1vf5 down
# ip link set pf1vf5 name pf1vf55
# ovs-vsctl -- --columns=name,ofport list Interface 
....
....                                                                                                                                                                                                                                                                                                                                                                                               
name                : pf1vf5                                                                                                                                                                  
ofport              : -1 
```